packages/table/src/TableContent.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { Table as RsTable } from 'reactstrap';
+import Icon from '@availity/icon';
+import TableHeader from './TableHeader';
+import TableHeaderRow from './TableHeaderRow';
+import TableHeaderCell from './TableHeaderCell';
+import TableRow from './TableRow';
+import TableCell from './TableCell';
+import { Cell, ExtendedTableHeader, IdType, Row, TableInstance, RowProps } from './types/ReactTable';
+import { useTableContext } from './TableContext';
+
+const TableContent = <T extends IdType>(): JSX.Element | null => {
+  const {
+    instance: tableInstance,
+    id,
+    scrollable,
+    sortable,
+    tableProps,
+    headerProps,
+    onSort,
+    bodyProps,
+    paged,
+    selectable,
+    AdditionalContent,
+    additionalContentProps,
+
+    onRowClick,
+    onCellClick,
+    getCellProps,
+    getRowProps,
+  } = useTableContext();
+
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow, manualSortBy, page } =
+    tableInstance as TableInstance<T>;
```

Comment:
Believe me, I have tried....many times. I have researched how to get Context/Provider to be generic and it is a mess. Because TableContext is static we have to know what type it is before it gets created.
The closest thing that I have found to working is this: https://hipsterbrown.com/musings/musing/react-context-with-generics/ - but it assumes that when you create the Provider it knows it's concrete type (not the generic type). 

This is the cleanest thing that I have found that works.